### PR TITLE
Refactor codebase to use the loggers property instead of logger

### PR DESCRIPTION
**Pull request by akashkw**
## What does this PR do?
Refactors the codebase to use the new loggers property introduced here https://github.com/PyTorchLightning/pytorch-lightning/pull/11683.

Part of #11232

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -580,18 +581,22 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
...
                 # the user has changed weights_save_path, it overrides anything
                 save_dir = trainer.weights_save_path
             else:
-                save_dir = trainer.logger.save_dir or trainer.default_root_dir
-
+                if len(trainer.loggers) == 1 and trainer.loggers[0].save_dir:
+                    save_dir = trainer.loggers[0].save_dir
+                else:
+                    save_dir = trainer.default_root_dir
+            # TODO: Find out we handle trainer.logger.version without LoggerCollection
```
Comment (akashkw) [inline pytorch_lightning/callbacks/model_checkpoint.py:593]: How should we handle trainer.logger.version and trainer.logger.name here? LoggerCollection will concatenate the values together from each logger, but with trainer.loggers we no longer have that logic and that logic might not be the best approach anyways.

File: pytorch_lightning/callbacks/progress/base.py [hunk 2]
```diff
@@ -213,6 +213,7 @@ def get_standard_metrics(trainer: "pl.Trainer", pl_module: "pl.LightningModule")
     if pl_module.truncated_bptt_steps > 0:
         items_dict["split_idx"] = trainer.fit_loop.split_idx
 
+    # TODO: Find out we handle trainer.logger.version without LoggerCollection
```
Comment (akashkw) [inline pytorch_lightning/callbacks/progress/base.py:216]: Same question as https://github.com/PyTorchLightning/pytorch-lightning/pull/11731/files#r801157300

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 3]
```diff
@@ -580,18 +581,22 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
...
             version = (
                 trainer.logger.version
                 if isinstance(trainer.logger.version, str)
                 else f"version_{trainer.logger.version}"
             )
+            # TODO: Find out we handle trainer.logger.name without LoggerCollection
```
Comment (akashkw) [inline pytorch_lightning/callbacks/model_checkpoint.py:599]: Same question as https://github.com/PyTorchLightning/pytorch-lightning/pull/11731/files#r801157300

File: pytorch_lightning/trainer/trainer.py [hunk 4]
```diff
@@ -2096,15 +2098,12 @@ def model(self, model: torch.nn.Module) -> None:
 
     @property
     def log_dir(self) -> Optional[str]:
-        if self.logger is None:
-            dirpath = self.default_root_dir
-        elif isinstance(self.logger, TensorBoardLogger):
-            dirpath = self.logger.log_dir
-        elif isinstance(self.logger, LoggerCollection):
+        if len(self.loggers) != 1 or isinstance(self.logger, LoggerCollection):
```
Comment (akashkw) [inline pytorch_lightning/trainer/trainer.py:2101]: When creating a LoggerCollection of size 1, the log_dir property returns default_root_dir, however we can't easily substitute this with trainer.loggers because trainer.loggers can't distinguish between a list of length 1 and a single logger. Both are stored as a list of length 1. Currently I'm checking trainer.logger, but this won't work once trainer.logger and LoggerCollection are deprecated. Should this be handled differently?

File: tests/trainer/properties/test_log_dir.py [hunk 5]
```diff
@@ -119,6 +119,7 @@ def test_logdir_logger_collection(tmpdir):
         max_steps=2,
         logger=[TensorBoardLogger(save_dir=save_dir, name="custom_logs")],
     )
+    # TODO: What behavior do we want once trainer.logger is deprecated?
```
Comment (akashkw) [inline tests/trainer/properties/test_log_dir.py:122]: How should this unit test be handled? Part of this issue https://github.com/PyTorchLightning/pytorch-lightning/pull/11731/files#r801162619.

File: tests/profiler/test_profiler.py [hunk 6]
```diff
@@ -495,6 +495,7 @@ def look_for_trace(trace_dir):
     # Wrap the logger in a list so it becomes a LoggerCollection
     logger = [TensorBoardLogger(save_dir=tmpdir)]
     trainer = Trainer(default_root_dir=tmpdir, profiler="pytorch", logger=logger, limit_train_batches=5, max_epochs=1)
+    # TODO: What behavior do we want once trainer.logger is deprecated?
```
Comment (akashkw) [inline tests/profiler/test_profiler.py:498]: How should this unit test be handled? Part of this issue https://github.com/PyTorchLightning/pytorch-lightning/pull/11731/files#r801162619
